frontend/src/pages/ProjectSettings/GitHubSettingsModal/GitHubSettingsModal.tsx

```diff
@@ -121,90 +160,154 @@ const GithubSettingsForm = ({
-							form.setValue(form.names.githubRepo, null)
-						}
-						disabled={!form.values.githubRepo}
-					>
-						Disconnect
-					</Button>
-				</Box>
+			<Box px="12" py="8" gap="12" display="flex" flexDirection="column">
+				<Form.NamedSection
+					label="Select GitHub repository"
+					name="githubRepo"
+				>
+					<Box display="flex" alignItems="center" gap="8">
+						<Select
+							aria-label="GitHub repository"
+							className={styles.repoSelect}
+							placeholder="Search repos..."
+							onSelect={(repo: string) =>
+								form.setValue(form.names.githubRepo, repo)
+							}
+							value={form.values.githubRepo?.split('/').pop()}
+							options={githubOptions}
+							notFoundContent={<span>No repos found</span>}
+							optionFilterProp="label"
+							filterOption
+							showSearch
+						/>
+						<ButtonIcon
+							kind="secondary"
+							emphasis="medium"
+							size="medium"
+							disabled={!form.values.githubRepo}
+							onClick={() =>
+								form.setValue(form.names.githubRepo, null)
+							}
+							icon={
+								<IconSolidTrash
+									size={16}
+									color={
+										vars.theme.interactive.fill.secondary
+											.content.text
+									}
+								/>
+							}
+						/>
+					</Box>
+				</Form.NamedSection>
 				{form.values.githubRepo && (
 					<>
-						<Text size="small">
-							Want to double check? Check it out on{' '}
-							<a
-								href={`https://github.com/${form.values.githubRepo}/`}
-								target="_blank"
-								rel="noreferrer"
-							>
-								GitHub
-							</a>
-							.
-						</Text>
+						<Box
+							display="flex"
+							gap="16"
+							paddingTop="12"
+							borderTop="dividerWeak"
+						>
+							<Form.Input
+								name={form.names.buildPrefix}
+								label="Build path prefix"
+								placeholder="/build"
+								icon={
+									<Tooltip
+										trigger={
+											<IconSolidQuestionMarkCircle
+												color={
+													vars.theme.static.content
+														.weak
+												}
+												size={14}
+											/>
+										}
+										renderInLine
+									>
+										<Box cssClass={styles.tooltipContent}>
+											The path added in the deployment
+											process to host your files, and
+											should be removed when trying to map
+											your files to GitHub.
+										</Box>
+									</Tooltip>
+								}
+							/>
+							<Form.Input
+								name={form.names.githubPrefix}
+								label="GitHub path prefix"
+								placeholder="/src"
+								icon={
+									<Tooltip
+										trigger={
+											<IconSolidQuestionMarkCircle
+												color={
+													vars.theme.static.content
+														.weak
+												}
+												size={14}
+											/>
+										}
+										renderInLine
+									>
+										<Box cssClass={styles.tooltipContent}>
+											The path removed in the deployment
+											process to host your files, which
+											can be seen in your GitHub repo.
```

Comment:
This one and the build path are definitely going to be hard to communicate but I think we can clean up the confusion with a visual on a docs page.